README.md

```diff
@@ -0,0 +1,29 @@
+# Haskell Docker Example
+
+This project is an example of producing a Docker image to run a Haskell program.
+It uses a two-stage approach to avoid the size overhead of a Haskell development
+environment in the runtime image.
+
+## Installation
+
+```
+make binaries
+make image
+```
+
+## Usage
+
+```
+docker run --rm example/hello
```

Comment:
I decided against that because, while it makes sense for "using" the example repository, it wouldn't make sense in projects based on this example -- they'd run locally via `stack`, deploy by pushing the image to a registry then _something else_ would ultimately run it (Heroku, AWS ECS, Kubernetes, etc).